libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -200,7 +204,15 @@ std::string IRGenerator::generate(
 	// Do not register immutables to avoid assignment.
 	t("DeployedObject", IRNames::deployedObject(_contract));
 	t("sourceLocationCommentDeployed", dispenseLocationComment(_contract));
-	t("library_address", IRNames::libraryAddressImmutable());
+	auto const eof = m_context.eofVersion().has_value();
+	t("eof", eof);
+	if (!eof)
+		t("library_address", IRNames::libraryAddressImmutable());
+	else
+		// TODO: We assume that libraries do not have immutable variables. The address is stored in 0 offset of
+		//  auxiliary data section. Is it correct?
```

Comment:
I think this is a correct assumption. Libraries cannot have state variables and this immutable for address is the only one I'm aware of.

I think that even if it's not correct, I think we'll easily find out soon enough because we'll have to deal with all the `loadimmutable` calls in the generator anyway so no need to worry too much.

Reply:
BTW, I think it might be best to deal with the immutable-related builtins first (`loadimmutable`, `setimmutable`). I see several options:
1. remove them from the dialect when EOF is enabled.
2. implement them on EOF using `auxdataload()`.
3. implement them on EOF, but make auxdataload()`'s 0 offset start after them.

Going with (2) or (3) would actually allow you not to care about it here at all. You'd just keep using the same immutable. You'd instead have to adjust the mechanism for allocating immutables to put them in aux data section.

I'm not sure whether (2) or (3) is preferable though.

Reply:
Ok. Option 1 is implemented here but I will investigate other options and we can decide which is the best.